src/powershell/Public/Remove-FinOpsCostExport.ps1

```diff
@@ -49,121 +42,83 @@ function Remove-FinOpsCostExport
-        -ResourceType "exports" `
-        -Name $Name `
-        -ApiVersion $ApiVersion `
-        -Method "DELETE" `
-        -Payload $payload
-
-      if ($httpResponse.StatusCode -eq 200)
-      {
-        # Export deleted successfully
-      }
-      elseif ($httpResponse.StatusCode -eq 404)
-      {
-        # Not found - nothing to delete
-      }
-      else
-      {
-        # Error response describing why the operation failed.
-        throw "Delete Cost Management Export operation failed with message: '$($httpResponse.Content)'"
-      }
-    }
-
-    # Delete associated ingestion data from storage account
-    if ($RemoveData)
-    {
-      # Using the REST API to get the export as requested as PS modules are outdated?
-      $httpResponse = Invoke-AzRestMethod `
-        -ResourceProviderName "Microsoft.CostManagement" `
-        -ResourceType "exports" `
-        -Name $Name `
-        -ApiVersion $ApiVersion `
-        -Method "GET" `
-        -Payload $payload
-
-      if ($httpResponse.StatusCode -eq 200)
+      # Delete associated data from storage account
+      if ($RemoveData)
       {
-
         # Export details retreived
         $exportDetails = ConvertFrom-Json -InputObject $httpResponse.Content
-        $storageAccountID = $exportDetails.properties.deliveryInfo.destination.resourceId
-
-        # Get-AzStorageAccount -resourcegroupname alz-finopstk -name finopshub255k7ov6asado | Get-AzDataLakeGen2ChildItem -FileSystem "ingestion" -Path "/" -Recurse -FetchProperty | Where-Object IsDirectory -ne false | Select-Object Path
+        $storageAccountId = $exportDetails.properties.deliveryInfo.destination.resourceId
         $resourceGroupName = $storageAccountID.Split('/')[4]
         $storageAccountName = $storageAccountID.Split('/')[8]
+        $path = $scope + "/" + $Name
 
-        #Hold on to your hats, deleting all the files in the ingestion scope. Using Az PS module as this should be current.
-        if ($PSCmdlet.ShouldProcess($scope, 'DeleteCostReports'))
+        if ($PSCmdlet.ShouldProcess($path, 'DeleteCostReports'))
         {
-          Write-Verbose "Resourace Group: $resourceGroupName"
-          Write-Verbose "Storage Account: $storageAccountName"
+          Write-Verbose "Resource group: $resourceGroupName"
+          Write-Verbose "Storage account: $storageAccountName"
           Write-Verbose "Scope: $scope"
 
-          $getFiles = Get-AzStorageAccount -ResourceGroupName $resourceGroupName -Name $storageAccountName | Get-AzDataLakeGen2ChildItem -FileSystem "ingestion" -Path $scope -Recurse -FetchProperty
-          if ($getFiles.Count -gt 0)
+          $getSta = Get-AzStorageAccount -ResourceGroupName $resourceGroupName -Name $storageAccountName
+          if ($getSta -and $getSta.EnableHierarchicalNamespace)
           {
-            $getFiles | Remove-AzDataLakeGen2Item -Force
+            $getSta | Remove-AzDataLakeGen2Item -FileSystem "msexports" -Path $path -Force
```

Comment:
Need to use the container associated with the export since the goal here is to delete exports that aren't connected to a hub instance.
```suggestion
            $getSta | Remove-AzDataLakeGen2Item -FileSystem $exportDetails.properties.deliveryInfo.destination.container -Path $path -Force
```